zipline/finance/blotter.py

```diff
@@ -227,10 +232,9 @@ def get_transactions(self, bar_data):
         closed_orders = []
         transactions = []
 
-        for asset, asset_orders in iteritems(self.open_orders):
-            trade_bar = bar_data[asset]
-
-            for order, txn in self.slippage_func(trade_bar, asset_orders):
+        for sid, asset_orders in iteritems(self.open_orders):
+            asset = self.asset_finder.retrieve_asset(sid)
```

Comment:
This is making a SQL Query per order.  You probably want `retrieve_all` to do this in one query.